### PR TITLE
[DROOLS-7067] Drools 8 docs : Default to Antora author mode

### PR DESCRIPTION
**Pull request by tkobayas**
- Fixed package-lock.json to make search-box work

**JIRA**: 
https://issues.redhat.com/browse/DROOLS-7067

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] tests</b>

- for a <b>full downstream build</b> 
  - for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

- <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

- <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] native</b>

- for <b>mandrel checks</b>  
  Run native checks against Mandrel image
  Please add comment: <b>Jenkins run mandrel</b>

- for a <b>specific mandrel check</b>  
  Run native checks against Mandrel image  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] mandrel</b>
</details>

### CODE REVIEW

**Comment by tkobayas** [comment]
- Now we don't need to specify `author` profile to run mvn. Just running mvn will build Antora site using local resources
- Now `src/antora.yml` is listed in `.gitignore`. `src/antora.yml` is created during maven build so no need to manage in git
- `package-lock.json` change (= just copying from optaplanner-website) makes search-box work. The issue seemed to be related to lunr version (2.3.8 vs 2.3.9).

**Comment by tkobayas** [comment]
@tarilabs @mariofusco Please review, thanks!

File: drools-docs/package.json [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
-    "@antora/cli": "3.0.0",
-    "@antora/site-generator": "3.0.0"
-  },
+  "name": "drools-docs",
+  "homepage": ".",
+  "private": true,
+  "license": "Apache-2.0",
   "dependencies": {
+    "@antora/cli": "^3.0.0",
+    "@antora/site-generator": "^3.0.0",
```
Comment (tarilabs) [inline drools-docs/package.json]: I admit I have not understood this change, sorry.

I would expect that the generated website does not need to depend on cli, or site-generator.
Could you check if it's really correct to have all these 3 as project dependency and not devDependency?

I'm fine either way, just reporting on what I've noticed :)

**Review by tarilabs — COMMENTED** [review]
good with me, I just have one doubt, below.

Thanks Toshiya!!

File: drools-docs/package.json (hunk 1 above)
Comment (tkobayas) [reply drools-docs/package.json]: @tarilabs Thank you for pointing that out! I copied the part from optaplanner-website without much thinking, sorry. I agree that cli and site-generator should be devDependency. Fixed.

**Review by mariofusco — APPROVED** [review]

**Comment by kie-ci3** [comment]
**(tests) - kogito-apps job** `#274` was: **UNSTABLE**
Possible explanation: This should be test failures

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-apps/274/display/redirect

**Test results:**
- PASSED: 1847
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-apps/274/testReport/org.kie.kogito.index/ProcessDataIndexMongoDBIT/testProcessInstanceEvents/">org.kie.kogito.index.ProcessDataIndexMongoDBIT.testProcessInstanceEvents</a></summary>
java.lang.AssertionError: <br/>1 expectation failed.<br/>JSON path data.UserTaskInstances[0].comments.size() doesn't match.<br/>Expected: is <1><br/>  Actual: <0><br/><br/>
</details>

File: drools-docs/package.json (hunk 1 above)
Comment (tarilabs) [reply drools-docs/package.json]: 👍

**Pull request merged**